src/hooks/useYValue.tsx

```diff
@@ -31,13 +43,28 @@ export function useYValue<T>(path: string): [
 
   // Initialization callback function
   const initialize = useCallback((yRoot: Y.Map<unknown>): void => {
-    const [v, p] = getValueByYPath(yRoot, yPath)
-    setValue(isYXmlText(v) ? v.toJSON() as T : v as T)
+    let vp = getValueByYPath(yRoot, yPath)
+
+    if (synced && vp[0] === undefined && options.createOnEmpty) {
+      // No value exists and caller wants us to fill in empty structure
+      const { path, data } = options.createOnEmpty
+      if (createYStructure(yRoot, path, data)) {
+        vp = getValueByYPath(yRoot, yPath)
+      }
+    }
+
+    if (options.observe) {
+      // When observing we return Y.XmlText as string
+      setValue(isYXmlText(vp[0]) ? vp[0].toJSON() as T : vp[0] as T)
+    } else {
+      // Just return raw Y values when not observing
+      setValue(vp[0] as T)
+    }
 
-    if (isYContainer(p)) {
-      setParent(p)
+    if (isYContainer(vp[1])) {
+      setParent(vp[1])
```

Comment:
I'm not too found of this, I'd rather see something along the line as the previous (and even more verbose):
`const [value, path] = getValueByPath(...)`

No one char variables, and no magic numbers.

Though I see now that you're reassigning it, which kind makes it harder.
I'm not too found of reassigning either :wink:

Reply:
Good points. I Agree. I'll refactor it!